apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -119,9 +119,18 @@ where
                             ),
                         }
                     } else {
+                        // If the public key corresponds to the MASP sentinel
+                        // transaction key, then the fee payer is effectively
+                        // the MASP, otherwise derive
+                        // they payer from public key.
+                        let fee_payer = if tx.pk != masp_tx_key().ref_to() {
```

Comment:
is it possible to use the sentinel key with non-MASP tx or is that checked somewhere?

Reply:
You're right, this should be checked somewhere. I'd been hoping that @juped 's validation code, once integrated, would do the rejection of sentinel signed transactions that either lack a shielded component or contain an invalid one.